clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4975,12 +4975,15 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   if (Args.hasArg(options::OPT_fclangir_direct_lowering))
     CmdArgs.push_back("-fclangir-direct-lowering");
 
+  if (Args.hasArg(options::OPT_fno_clangir_direct_lowering))
+    CmdArgs.push_back("-fno-clangir-direct-lowering");
+
```

Comment:
This should be equivalent, and also handle cases where both the positive and negative flag are specified (last one wins). I'd add a test case for that too (with both `-fclangir-direct-lowering -fno-clangir-direct-lowering` and `-fno-clangir-direct-lowering -fclangir-direct-lowering`).

```suggestion
  Args.addOptOutFlag(CmdArgs, options::OPT_fclangir_direct_lowering, options::OPT_fno_clangir_direct_lowering)
```

Reply:
Good point!